app/src/lib/stores/app-store.ts

```diff
@@ -1609,7 +1546,51 @@ export class AppStore extends TypedBaseStore<IAppState> {
       return { workingDirectory, selectedFileIDs, diff }
     })
 
-    this._triggerMergeConflictsFlow(repository, status)
+    this.repositoryStateCache.updateChangesState(repository, state => {
```

Comment:
can we pull this into its own method?

Reply:
Mind if I do this in a follow-up PR? There 's an existing [function above](https://github.com/desktop/desktop/blob/6d0206acc9f4ea543266acf948cf5064260f96e2/app/src/lib/stores/app-store.ts#L1488-L1546) that is also rather lengthy, and I'd love to tackle both of those at the same time.